weave/trait/pairing/basic_open_close_installation_state_trait.proto

```diff
@@ -0,0 +1,116 @@
+/*
+ *
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file specifies the basic open close installation state trait.
+ */
+
+syntax = "proto3";
+
+package weave.trait.pairing;
+
+import "wdl/wdl_options.proto";
+
+option java_outer_classname = "WeaveInternalBasicOpenCloseInstallationStateTrait";
+option objc_class_prefix = "SCM";
+
+/**
+ *  @brief
+ *    Basic Open Close Installation State Trait
+ *
+ *  A Weave Common service published trait that provides a means for an
+ *  application to set a an open close sensor device to different modes
```

Comment:
"provides a means for an application to facilitate setup and installation of a basic reed-switch based open close sensor. This is achieved through a set of queryable installation modes that an application can usher the device through by setting them"

Reply:
Done.